src/DataBox/DataBox/Common/Cmdlets/GetDataBoxJobCmdlet.cs

```diff
@@ -0,0 +1,176 @@
+                                        jobPageList.NextPageLink);
+                    }
+
+                    
+                    if (Completed || Cancelled || Aborted || CompletedWithErrors)
+                    {
+                        foreach (var job in jobPageList)
+                        {
+                            if ((Completed && job.Status == StageName.Completed) 
+                                || (Cancelled && job.Status == StageName.Cancelled) 
+                                || (Aborted && job.Status == StageName.Aborted)
+                                || (CompletedWithErrors && job.Status == StageName.CompletedWithErrors))
+                            {
+                                result.Add(job);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        result.AddRange(jobPageList.ToList());
+                    }
+
+                } while (!(string.IsNullOrEmpty(jobPageList.NextPageLink)));
+                foreach(var job in result)
+                {
+                    finalResult.Add(new PSDataBoxJob(job));
+                }
+                WriteObject(finalResult, true);
+            }
+            else
+            {
+
+                 IPage<JobResource> jobPageList = null;
+                 List<JobResource> result = new List<JobResource>();
+                 List<PSDataBoxJob> finalResult = new List<PSDataBoxJob>();
+
+
+                do
+                {
+                     // Lists all the jobs available under the subscription.
+                     if (jobPageList == null)
+                     {
+                         jobPageList = JobsOperationsExtensions.List(
+                                         this.DataBoxManagementClient.Jobs);
+                     }
+                     else
+                     {
+                         jobPageList = JobsOperationsExtensions.ListNext(
+                                         this.DataBoxManagementClient.Jobs,
+                                         jobPageList.NextPageLink);
+                     }
+                    if (Completed || Cancelled || Aborted || CompletedWithErrors)
+                    {
+                        foreach (var job in jobPageList)
+                        {
+                            if ((Completed && job.Status == StageName.Completed) 
+                                || (Cancelled && job.Status == StageName.Cancelled) 
+                                || (Aborted && job.Status == StageName.Aborted)
+                                || (CompletedWithErrors && job.Status == StageName.CompletedWithErrors))
+                            {
+                                result.Add(job);
+                            }
+                        }
+                    }
+                    else
+                    {
+
+                        result.AddRange(jobPageList.ToList());
```

Comment:
Not that you can write these to the pipeline as you discover them.

Reply:
I want to write it once to the pipeline. Is it not okay?